docs/recap/bitvm2_fast_recap.tex

```diff
@@ -0,0 +1,229 @@
+\usepackage{hyperref}
+\makeatletter
+\newcommand{\citeprocitem}[2]{\hyper@linkstart{cite}{citeproc_bib_item_#1}#2\hyper@linkend}
+\makeatother
+\usepackage[a4paper, total={6in, 8in}]{geometry}
+\author{Kyrylo Baibula}
+\date{\today}
+\title{BitVM fast recap}
+\hypersetup{
+  pdfauthor={Kyrylo Baibula},
+  pdftitle={BitVM fast recap},
+  pdfkeywords={},
+  pdfsubject={},
+  pdfcreator={Emacs 29.4 (Org mode 9.7.11)},
+pdflang={English}}
+\usepackage{biblatex}
+
+\addbibresource{refs.bib}
+
+\begin{document}
+
+\maketitle
+
+\begin{abstract}
+  Assume some user wants to publicly execute a large program on the
+  Bitcoin chain using Bitcoin Script, but its implementation is larger
+  than 4 megabytes (for example, a Zero Knowledge proof); that is why it
+  cannot fit in the transaction input. This document is a fast recap of
+  the BitVM2 doc, which proposes optimistic execution verification with
+  fraud proofs in case of incorrectness.
+\end{abstract}
+
+\setcounter{tocdepth}{2}
+\tableofcontents
+
+\section{Introduction}\label{sec:intro}
+
+In the ever-evolving landscape of blockchain technology, the desire to
+execute increasingly complex and large-scale programs on the
+Bitcoin\autocite{bitcoin_paper} network is growing. However, Bitcoin
+Script, the native programming language of Bitcoin, imposes strict
+size limits, such as the 4-megabyte cap on transaction inputs, which
+makes it challenging to implement certain advanced cryptographic
+proofs, like Zero Knowledge proofs. To address this limitation, the
+BitVM2\autocite{bitvm2} proposal introduces an innovative approach
+that enables the optimistic execution of large programs on the Bitcoin
+chain. This method leverages fraud proofs to ensure the validity of
+executions, offering a robust mechanism for verifying computations,
+while providing a fail-safe against incorrect or malicious
+operations. This document provides a concise overview of BitVM2,
+highlighting its potential to unlock new possibilities for secure and
+scalable program execution on Bitcoin.
+
+\section{Program splitting}\label{sec:program-splitting}
+
+\begin{quote}
+  Let there be a large program \(f\), which can be described in Bitcoin
+  Script. We want to compute it \textbf{\textbf{on-chain}}, i.e., find
+  such an \(f(x) = y\). To achieve this, we divide the program into
+  \(n\) sub-programs \(f_1, \ldots, f_n\) and their corresponding
+  intermediate states \(z_1, \ldots, z_n\), such that:
+
+  \begin{equation}
+    \begin{aligned}
+      f_1(x) &= z_1 \\
+      f_2(z_1) &= z_2 \\
+      &\cdots \\
+      f_n(z_{n-1}) &= y
+    \end{aligned}
+  \end{equation}
+\end{quote}
+
+However, the user (referred in BitVM2 as the operator) only needs to
+prove that the given program \(f\) indeed returns \(y\) for \(x\), or
+\textbf{give others the opportunity} to \textbf{disprove this
+fact}. In our case, this means giving challengers the ability to
+prove that at least one of the sub-programs \(f_i(z_{i-1}) \neq z_i\).
```

Comment:
one of the sub-programs... what?

Reply:
Again, my bad

Reply:
My bad, the original statement was:

> довести, що хоча б для однієї з підпрограм виконується твердження $f_i(z_{i-1}) \neq z_i$

May be we should make it:

```suggestion
prove that for at least one of the sub-programs statement \(f_i(z_{i-1}) \neq z_i\) is true.
```

Reply:
Yeah, sounds good to me